### DISCUSSION TITLE
[MIP-7] Edition Print Authorities

### DISCUSSION THREAD

**Discussion by svc01-metaplex** [metaplex-foundation/mip#28] [MIP Submissions]
Submitter: meditatingsloth
Email: austin.borden@gmail.com
Submitter wallet: MADMQ67GBNsVQPtq1ZnYSZzJ3Fd2gLJFWtT6bKWo97E
Submission transaction: 5JUs1dp7Edw2MXNCpoeBcZMt9jVtSLJXwNMGd5NfqVqgapJYKVctuE8QUUPhtsxeUng8PWgnknCchpHi48qEPM8i
Title: [MIP-7] Edition Print Authorities
Links: 

# Overview

The Token Metadata program currently only supports printing editions from a master edition if the signer is also the owner of the token. This forces creators to escrow their master edition token when listing an edition sale on a marketplace, as the marketplace must be the owner of the token in order to print editions. This also prevents the creator from being able to list their master edition to be printed on multiple marketplaces concurrently.

# Motivation

The main motivation behind the proposal to add edition print authorities is to give creators the ability to list edition sales on multiple marketplaces concurrently. This gives creators more flexibility in how they choose to sell their work while remaining in control of their master edition. Marketplaces will also benefit from this feature as creators would no longer be forced to choose a single marketplace to list their work on.

# Proposal

It is our proposal to add a new `PrintAuthority` authority type, similar to how `CollectionAuthority` works, along with the instructions to approve and revoke authorities. A new `print_v2` instruction would also be required to print an edition using a verified print authority.

# Community Asks

- Are there any downsides to allowing multiple authorities to print from the same master edition?
- What other technical alternatives should we consider?
- Any other pertinent input?

**Comment by smkly** [comment]
plz fix

**Comment by BloodyBee** [comment]
I'm not enough of a chad to comment on the implementation path here but wanna 
voice my support and say that this is an important topic to tackle.

As an artist on Solana I'm deeply grateful to see this happen, on a personal level. 
But a solution to this problem will obviously greatly benefit the ecosystem at large.

So, more power to you! Let's make this happen.

**Comment by echohtp** [comment]
Metaplex team, what additional information is needed to get this proposal through initial review successfully?

Thanks for the feedback!

**Comment by nickfrosty** [comment]
I think this is a good idea and generally support it!
Food for thought: from what I can tell, the current `CollectionAuthority` implementation [does not support multiple](https://github.com/metaplex-foundation/mpl-token-metadata/blob/965b60860afdf5c73092563b83e534411d088327/clients/rust/src/generated/instructions/approve_collection_authority.rs#L118) authorities. Just a single one. So there would need to be a different implementation used to support multiple "print authorities"

**Reply by samuelvanderwaal** [reply]
mo accounts, mo problems

**Reply by meditatingsloth** [reply]
@nickfrosty 
When approving a collection authority the instruction looks to be creating a PDA keyed by the new authority address, so it does seem to support multiple authorities. https://github.com/metaplex-foundation/mpl-token-metadata/blob/965b60860afdf5c73092563b83e534411d088327/programs/token-metadata/program/src/processor/collection/approve_collection_authority.rs#L44 
Does that look right?

@samuelvanderwaal Is it better practice to reduce number of accounts? Would be great to get your thoughts. The number of accounts in this context would be constrained by the number of marketplaces the owner would want to print from. As far as printing goes the marketplace would only need to know about the PDA for their own listing, which can easily be derived.

**Reply by samuelvanderwaal** [reply]
For some reason I initially thought Nick was talking about supporting multi-sig signers. The current collection authority is a PDA with the delegate address as one of the seeds, so you can indeed have multiple collection delegates for any given collection. You could design the `PrintAuthority` similarly.

**Reply by nickfrosty** [reply]
I was not talking about multi-sig signers! 
But good to know the Print authority stuff would be more similar to the collection authority than I expected

**Comment by MetaplexFoundation** [comment]
The community feedback period for this submission has ended. It will now be converted to MIP-7 and move to a vote on the Metaplex DAO (https://dao.metaplex.com/). The voting period will open on November 15, 2023, and remain open for 14 days. If passed, MIP-7 will be a Metaplex Foundation implementation.

**Comment by MetaplexFoundation** [comment]
The Metaplex DAO voted to approve MIP-7. Development for MIP-7, including the accompanying independent security audit, is expected to be completed by the end of January 2024.

**Comment by blockiosaurus** [comment]
MIP-7 has been merged into mpl-token-metadata main branch and will be deployed to devnet for testing and audits shortly.
